modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaIndexerCheck.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringUtil;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class UpgradeJavaIndexerCheck
+	extends BaseUpgradeMatcherReplacementCheck {
+
+	@Override
+	protected String formatMatcherIteration(
+		String content, String newContent, Matcher matcher) {
+
+		String newLine = StringUtil.replace(
+			matcher.group(), "Indexer", "Indexer<?>");
+
+		return StringUtil.replace(newContent, matcher.group(), newLine);
```

Comment:
Hi @michaelmdelcavalcanti, as we discussed, please use `StringUtil.replace(matcher.group(), "Indexer", "Indexer<?>")` instead instantiate the `newLine` variable.